adapters/repos/db/lsmkv/segment_group_compaction.go

```diff
@@ -122,6 +124,31 @@ func (sg *SegmentGroup) compactOnce() (bool, error) {
 		return false, nil
 	}
 
+	if sg.allocChecker != nil {
```

Comment:
You can have a non-pointer field that has a default nullAllocChecker that does nothing, and then you don't need to insert nil schecks everywhere in the code, and also you won't add another nil pointer problem

Reply:
Agreed, this would also avoid having a lot of nil parameters being passed around, which can make it harder to read the code

Reply:
I tried this, but I dont think it works :(

The allocChecker is passed around a lot and we would replace the `if sg.allocChecker != nil` at usage with nil checks at struct initialization. From following a few code-paths that would result in more nil-checks and especially the possibility to forget it only for some code paths. I think having it at the usage place makes it easier to remember that it is optional